internal/proxy/task_query.go

```diff
@@ -378,9 +378,23 @@ func (t *queryTask) queryShard(ctx context.Context, leaders *querypb.ShardLeader
 	return nil
 }
 
-func (t *queryTask) checkIfLoaded(collectionID UniqueID, searchPartitionIDs []UniqueID) bool {
-	// If request to search partitions
-	if len(searchPartitionIDs) > 0 {
+func (t *queryTask) checkIfLoaded(collectionID UniqueID, queryPartitionIDs []UniqueID) bool {
+	// check if collection was loaded into QueryNode
+	info, err := globalMetaCache.GetCollectionInfo(t.ctx, t.collectionName)
+	if err != nil {
+		log.Warn("fail to check if collection is loaded",
+			zap.Int64("requestID", t.Base.MsgID),
+			zap.Int64("collectionID", collectionID),
+			zap.String("requestType", "query"),
+			zap.Error(err))
+		return false
```

Comment:
should not return false but return error.
Otherwise the error message would be very confusing

Reply:
you're right